bindings/experimental/distrdf/python/DistRDF/HeadNode.py

```diff
@@ -210,3 +229,94 @@ def build_ranges(self):
         logger.debug("%s clusters will be split along %s partitions.",
                      numclusters, self.npartitions)
         return Ranges.get_clustered_ranges(clustersinfiles, self.npartitions, self.friendinfo)
+
+    def generate_rdf_creator(self):
+        """
+        Generates a function that is responsible for building an instance of
```

Comment:
Perhaps instead of repeating all this for the two sources, we can say something shorter that is specific to a source. Like: `Generates a function that builds an instance of RDataFrame on a distributed mapper for a given entry range, specifically for the tree data source.`, and similarly for the empty DS. Perhaps the justification you detail below is not strictly necessary (it already makes sense that a node type knows how to generate the DF for that type).

Reply:
Ok good point!